src/net.cpp

```diff
@@ -474,16 +476,18 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
                 if (m_i2p_sam_session) {
```

Comment:
The function `AddI2PSessionsIfNeeded` is called without checking if I2P is enabled, which may lead to unnecessary session creation.